src/routes/imports/add.ts

```diff
@@ -0,0 +1,50 @@
+import Joi from 'joi';
+import { ControllerGeneratorOptionsWithClient } from '../../comms/controller';
+import {
+  ProcessingImport,
+  processingImportSchema,
+  InvalidImport,
+  invalidImportSchema,
+} from '../../models/imports';
+import { B64XLSXFile, b64XLSXFile } from './types';
+
+type Request = {
+  body: {
+    dataUrl: B64XLSXFile;
+    /**
+      * name override for the import.
+      * Defaults to filename
+      */
+    name?: string | null;
```

Comment:
1. Why do you want to make it optional?
2. Why are you allowed to pass `null` if we already can leave it out?

Reply:
2. True, I can remove this

1. The idea would be that the platform set the name following some rules, for example, defaults to the filename. But if the user wants a specific name, they can choose one.

Reply:
The filename is not included in the base64 I think? So the client would have to extract it (like it does [here](https://github.com/withthegrid/platform-client/blob/c10f323df5cbf4fe57500dbcbdd8c34e13b977d4/src/views/generic/file-upload.vue#L217-L218)) and feed it as name parameter. I think that name should be `string` only..